lib/rules/no-implicit-globals.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * @fileoverview Rule to check for implicit global variables and functions.
+ * @author Joshua Peek
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+
+    return {
+        "Program": function() {
+            var scope = context.getScope();
```

Comment:
This scope stuff is pretty cool. It basically handled all the module scoping cases. :metal: